OpenRA.Mods.Common/Widgets/Logic/ReplayUtils.cs

```diff
@@ -0,0 +1,40 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.FileFormats;
+
+namespace OpenRA.Mods.Common.Widgets.Logic
+{
+	public static class ReplayUtils
+	{
+		static readonly Action DoNothing = () => { };
+
+		public static bool CheckReplayCompatibility(ReplayMetadata replayMeta, Action onConfirm = null, Action onCancel = null)
+		{
+			if (onConfirm == null)
+				onConfirm = DoNothing;
+
+			if (onCancel == null)
+				onCancel = DoNothing;
+
+			if (replayMeta.GameInfo.Mod != Game.ModData.Manifest.Mod.Id)
+				ConfirmationDialogs.PromptConfirmAction("Incompatible replay", "This replay was recorded with a different mod {0}.".F(replayMeta.GameInfo.Mod), onConfirm, onCancel);
+			else if (replayMeta.GameInfo.Version != Game.ModData.Manifest.Mod.Version)
+				ConfirmationDialogs.PromptConfirmAction("Incompatible replay", "This replay was recorded with a different engine version {0}.".F(replayMeta.GameInfo.Version), onConfirm, onCancel);
+			else if (replayMeta.GameInfo.MapPreview.Status != MapStatus.Available)
+				ConfirmationDialogs.PromptConfirmAction("Incompatible replay", "This replay was recorded with an unavailable map {0}.".F(replayMeta.GameInfo.MapUid), onConfirm, onCancel);
```

Comment:
These strings need a line break somewhere, they are too wide for the dialog box.